hack/release-tool.py

```diff
@@ -0,0 +1,129 @@
+
+    return f'{ns}{arch_suffix}{private_suffix}', f'{name}{arch_suffix}{private_suffix}'
+
+
+def validate_server_connection(ctx):
+    with oc.options(ctx), oc.tracking(), oc.timeout(60):
+        try:
+            username = oc.whoami()
+            version = oc.get_server_version()
+            logger.debug(f'Connected to APIServer running version: {version}, as: {username}')
+        except (ValueError, OpenShiftPythonException, Exception) as e:
+            logger.error(f"Unable to verify cluster connection using context: \"{ctx['context']}\"")
+            raise e
+
+
+def update(ctx, action, ns, name, release, custom_message, custom_reason, execute):
+    with oc.options(ctx), oc.tracking(), oc.timeout(5*60):
+        try:
+            with oc.project(ns):
+                tag = oc.selector(f'imagestreamtag/{name}:{release}').object(ignore_not_found=True)
+                if not tag:
+                    raise Exception(f'Unable to locate imagestreamtag: {ns}/{name}:{release}')
+
+                ts = int(round(time.time() * 1000))
+                backup_filename = f'{name}_{release}-{ts}.json'
+                if execute:
+                    with open(backup_filename, mode='w+', encoding='utf-8') as backup:
+                        logger.debug(f'Creating backup file: {backup_filename}')
+                        backup.write(json.dumps(tag.model._primitive(), indent=4))
+
+                def update_annotations(obj):
+                    message = f'Manually {action}ed per TRT'
+                    if custom_message is not None:
+                        message = custom_message
+
+                    for annotations in (obj.model.image.metadata.annotations, obj.model.metadata.annotations, obj.model.tag.annotations):
+                        if action == 'accept':
+                            annotations['release.openshift.io/phase'] = 'Accepted'
+                        elif action == 'reject':
+                            annotations['release.openshift.io/phase'] = 'Rejected'
+                        else:
+                            raise ValueError(f'Unsupported action specified: {action}')
+
+                        annotations['release.openshift.io/message'] = message
+
+                        if custom_reason is not None:
+                            annotations['release.openshift.io/reason'] = custom_reason
+
+                    logger.info(f'{action.capitalize()}ing: {ns}/{name}:{release}')
+                    if execute:
+                        logger.debug(f'Updated payload:\n{json.dumps(obj.model._primitive(), indent=4)}')
+                        return True
+                    else:
+                        logger.info(f'Updated payload:\n{json.dumps(obj.model._primitive(), indent=4)}')
+                        logger.warning('You must specify "--execute" to permanently apply these changes')
+                        exit(0)
+
+                update_annotations(tag)
+                tag.replace()
```

Comment:
Reading, changing, then calling UPDATE may stomp other data that's been changed in the interim or give you an "already modified" error, depending on the setup. Prefer generating a merge patch or using server-side apply.

Reply:
Converted to using `patch()`